### PR TITLE
Feature/core 103/fix mcq exits

### PR DESCRIPTION
**Pull request by bulatgab**
ezgif com-video-to-gif

### CODE REVIEW

File: src/components/interaction-designer/block-editors/choices/ChoicesBuilder.vue [hunk 1]
```diff
@@ -143,39 +143,38 @@ export class ChoicesBuilder extends mixins(Lang) {
...
-        console.debug('delete previous empty choice at index', previousIndex)
-        this.deleteChoiceByResourceIdFrom({blockId: this.block.uuid, resourceId: previousLastChoice.prompt})
-        // important to delete the choice before removing the resource, otherwise this will throw an error
-        this.flow_removeResourcesAndRelatedValidationsOnActiveFlow({
-          resourceUuids: [resourceId],
-        })
-        previousIndex -= 1
-        previousLastChoice = this.block.config.choices[previousIndex]
-      }
+      this.deleteLastAndPreviousChoicesIfEmpty(choiceIndex)
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/choices/ChoicesBuilder.vue:146]: extracted this into a separate function

File: src/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ const actions: ActionTree<IEmptyState, IRootState> = {
       // create new exit if it doesn't exist yet
       if (exitsForChoices[i] == null) {
         const uuid = await (new IdGeneratorUuidV4()).generate()
-        const exit = await dispatch('flow/block_createBlockExitWith', {props: {uuid} as IBlockExit}, {root: true})
+        const exit: IBlockExit = await dispatch('flow/block_createBlockExitWith', {props: {uuid} as IBlockExit}, {root: true})
```
Comment (bulatgab) [inline src/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore.ts:79]: this fixes a type warning on the next line

File: src/store/validation/index.ts [hunk 3]
```diff
@@ -353,6 +354,7 @@ export const validationActions: ActionTree<IValidationState, IRootState> = {
       ?.map(error => extractResourceVariantIndex(error.dataPath))
       .filter(isNotUndefined)
       .map(invalidVariantIndex => resource.values[invalidVariantIndex])
+      .filter(invalidVariant => invalidVariant.mime_type !== getters.choiceMimeType)
```
Comment (bulatgab) [inline src/store/validation/index.ts:357]: This removes invalid choice resources from 'invalid resources' - we don't care about them on the resource viewer page.

It was giving a console error:
Screenshot 2023-02-15 at 10 25 36 PM
when there were empty choices in the middle.

**Review by object-required — APPROVED** [review]

File: src/components/interaction-designer/block-editors/choices/ChoicesBuilder.vue (hunk 1 above)
Comment (safydy-r) [reply src/components/interaction-designer/block-editors/choices/ChoicesBuilder.vue:146]: Thanks @bulatgab

**Review by safydy-r — APPROVED** [review]
neat PR !

**Pull request merged**